composables/drop/massmint/useDropMassMint.ts

```diff
@@ -0,0 +1,233 @@
+import {
+  AllocatedNFT,
+  DoResult,
+  allocateCollection,
+  allocateClaim as claimAllocation,
+} from '@/services/fxart'
+import useGenerativePreview, {
+  EntropyRange,
+  GenerativePreviewItem,
+} from '../useGenerativePreview'
+import { ImageDataPayload } from '../useGenerativeIframeData'
+import { ToMintNft } from '@/components/collection/drop/types'
+import { getFakeEmail } from '@/components/collection/drop/utils'
+import useDropMassMintPreview from './useDropMassMintPreview'
+import useDropMassMintUploader from './useDropMassMintUploader'
+import { useCollectionEntity } from '../useGenerativeDropMint'
+
+export type MassMintNFT = Omit<ToMintNft, 'priceUSD'> & {
+  imageDataPayload?: ImageDataPayload
+  metadata?: string
+  hash: string
+  sn?: number
+  entropyRange: EntropyRange
+}
+
+type MassMintParams = {
+  isLoading: Ref<boolean>
+}
+
+export default ({ isLoading }: MassMintParams) => {
+  const { accountId } = useAuth()
+
+  useDropMassMintUploader()
+  const { generatePreviewItem } = useGenerativePreview()
+  const { allPinned, payloads, pinMetadata, getPreviewItemsToMintedNfts } =
+    useDropMassMintPreview()
+  const { mintedAmountForCurrentUser } = useCollectionEntity()
+
+  const {
+    drop,
+    amountToMint,
+    previewItem,
+    allocatedNFTs,
+    toMintNFTs,
+    mintingSession,
+    mintedNFTs,
+  } = storeToRefs(useDropStore())
+
+  const isAllocating = ref(false)
```

Comment:
usefull how?

it isn't being returned from the composbale

Reply:
removed